### ISSUE TITLE
How to get rid of Wpointer-to-int-cast warnings

### ISSUE THREAD

**Issue by iongion** [rahra/cairo_jpg#1]
Can these warnings be avoided ?

```
cairo_jpg.c:223:15: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]

  return write((long)closure, data, length) < length ? CAIRO_STATUS_WRITE_ERROR : CAIRO_STATUS_SUCCESS;

cairo_jpg.c: In function 'cairo_image_surface_write_to_jpeg':
cairo_jpg.c:279:62: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]

  e = cairo_image_surface_write_to_jpeg_stream(sfc, cj_write, (void *)(long)outfile, quality);
```

**Comment by psychon** [comment]
The code should use `intptr_t` instead of `long`.
@iongion Do you happen to compile this on Windows on a 64-bit machine? (Or why does your `long` have a different size than `void*`?)

**Comment by iongion** [comment]
@psychon Yes, you are right, I am compiling on msys (https://www.msys2.org/) running in Windows 64 bit

**Comment by rahra** [comment]
Interesting Windows behavior.
Did you find a solution?
I'll wrap it into an #ifdef macro.

**Comment by psychon** [comment]
@rahra Just always use `intptr_t` everywhere instead of `long`. No `#ifdef` needed (just a recent-enough C environment and perhaps `stdint.h`).

**Comment by rahra** [comment]
I fixed the type conversions in 69a2b785144faedbb36f19557c02aac8ea6bf46a.
